spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -424,6 +424,15 @@ public final void setTransactionIdPrefix(String transactionIdPrefix) {
 		return this.transactionIdPrefix;
 	}
 
+	public final boolean initTransactionIdSuffix(int transactionIdSuffix) {
+		Assert.isTrue(transactionIdSuffix >= 0, "'transactionIdSuffix' initial value must greater than or equal 0");
+		return this.transactionIdSuffix.compareAndSet(0, transactionIdSuffix);
+	}
```

Comment:
Hi, @sobychacko new interface is here. `DefaultKafkaProducerFactory` use only

Reply:
Thank you, but I didn't quite mean that, so I've decided to lend a hand. Could you please take a [look](https://github.com/spring-projects/spring-kafka/pull/2930)?